src/lib/parser/statement/parse.rs

```diff
@@ -219,6 +219,18 @@ pub(crate) fn parse(code: &str) -> Statement {
             return Statement::Time(Box::new(parse(cmd[4..].trim_left())))
         }
         _ if cmd.eq("time") => return Statement::Time(Box::new(Statement::Default)),
+        _ if cmd.starts_with("and ") => {
+            return Statement::And(Box::new(parse(cmd[3..].trim_left())))
+        }
```

Comment:
I wonder if we could make a macro to simplify how this is currently done here

Reply:
Basically, rather than forcing an allocation here to make it an owned string, you could alter the return type so that it returns `Statement`s instead of `String`s, which will require some changes to fix tests and such.

Reply:
After having a look I don't think it changes much and still doesn't solve the main problem I was having in the first place with `data: &str` was figuring out how to replace the data field with the modified statement string and I couldn't seem to navigate my way through all the compiler errors...

Reply:
Basically, there's no need to modify the string. You can go ahead with converting it directly into a Statement, rather than returning an owned string to later be converted into a Statement.

Reply:
I'll likely go ahead and merge this and fix it soon.